controller/consoleapplication_controller.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	appsv1alpha1 "github.com/openshift-console/console-application-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	gitservice "github.com/openshift-console/console-application-operator/pkg/git-service"
+)
+
+// ConsoleApplicationReconciler reconciles a ConsoleApplication object
+type ConsoleApplicationReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=apps.console.dev,resources=consoleapplications,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps.console.dev,resources=consoleapplications/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=apps.console.dev,resources=consoleapplications/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *ConsoleApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling ConsoleApplication...")
+
+	// Retrieving the ConsoleApplication CR instance requested for reconciliation
+	operatorCR := &appsv1alpha1.ConsoleApplication{}
+	if err := r.Get(ctx, req.NamespacedName, operatorCR); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Error(err, "ConsoleApplication Not Found")
+			return NoRequeue()
+		}
+		logger.Error(err, "Unable to fetch ConsoleApplication CR")
+		SetDegraded(operatorCR, appsv1alpha1.ReasonOperatorResourceNotAvailable.String(), err.Error())
+		if err := r.Status().Update(ctx, operatorCR); err != nil {
+			return RequeueWithError(err)
+		}
+		return RequeueOnError(err)
+	}
+
+	init := operatorCR.Status.Conditions == nil
+	if init {
+		operatorCR.Status.Conditions = make([]metav1.Condition, 0)
+		SetStarted(operatorCR)
+		if err := r.Status().Update(ctx, operatorCR); err != nil {
+			return RequeueWithError(err)
+		}
```

Comment:
not sure I understand why updating the status here ? There's no condition you are setting . Are you resetting the conditions ? In that case I don't think it's needed. We can keep those there and update them only.

Reply:
I think it seems that I am creating an empty condition slice and updating the status with that. But the `SetStarted` function is adding a status condition "Progressing". 

Does this work? Or am I not getting it?

Reply:
Due to the complexities of having the "Progressing" status condition, I have currently moved forward with having solely the "Ready" status condition to inform users whether the application is created or is it in processing.